edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/pvc/KubernetesPvcMapper.cs

```diff
@@ -56,18 +58,23 @@ V1PersistentVolumeClaim ExtractPvc(KubernetesModule module, Mount mount, IDictio
                     Requests = new Dictionary<string, ResourceQuantity>() { { "storage", new ResourceQuantity($"{this.persistentVolumeClaimSizeMb}Mi") } }
                 },
             };
-
-            // prefer persistent volume name to storage class name, if both are set.
             if (this.persistentVolumeName.HasValue)
             {
-                persistentVolumeClaimSpec.VolumeName = this.persistentVolumeName.OrDefault();
+                string volumeNameTmp = this.persistentVolumeName.OrDefault();
+                if (volumeNameTmp != volumeName)
+                {
+                    throw new InvalidModuleException(string.Format("The mount name {0} has to be the same as the PV name {1}", volumeName, volumeNameTmp));
+                }
             }
-            else if (this.storageClassName.HasValue)
+
+            persistentVolumeClaimSpec.VolumeName = volumeName;
```

Comment:
Is it expected to change here? it used to ignore given VolumeName when `persistentVolumeName=Some(name)`. Not sure I understood changed logic here.

Reply:
This works fine for when PersistentVolume is specified, but this isn't working for the storage class case - PVC is still pending.

Reply:
OK